lib/vega_lite/data.ex

```diff
@@ -327,6 +327,381 @@ defmodule VegaLite.Data do
+    * `:y` - options for the y-axis
+    * `:color` - options for the color field. If given, the field specified will
+      be used as the grouping for the plot legend.
+
+  ## Examples
+
+  In this first example, we plot a standalone cartesian plot.
+  Note that we're configuring the X and Y axes of the plot through the `fields` argument.
+
+      data = %{
+        "r" => [1, 2, 3, 3, 4],
+        "theta" => [0, 30, 45, 135, 270],
+        "Group" => List.duplicate("First Line", 5)
+      }
+
+      VegaLite.Data.polar_plot(
+        VegaLite.new(height: 500, width: 500),
+        data,
+        [type: :line, point: true, interpolate: :cardinal, color: "black"],
+        r: "r",
+        theta: "theta",
+        x: [scale: [domain: [-5, 5]]],
+        y: [scale: [domain: [-10, 3]]],
+        color: "Group"
+      )
+
+  In this second example, we plot data onto a polar grid. Note that we include include the points
+  as separate layers for more customization.
+
+      color_key = "Line Groups"
+
+      data = %{
+        "r" => [1, 2, 3, 3, 4],
+        "theta" => [0, 30, 45, 135, 270],
+        color_key => List.duplicate("First Line", 5)
+      }
+
+      other_data = %{
+        "r" => [8, 6, 4, 2, 0],
+        "theta" => [0, 30, 60, 90, 120],
+        color_key => List.duplicate("Second Line", 5)
+      }
+
+      list = [
+        {data, type: :line, point: true, interpolate: "cardinal"},
+        {data, :point},
+        {other_data, type: :line, point: true, interpolate: "cardinal"},
+        {other_data, type: :point, stroke: "black", stroke_width: 2, size: 100, shape: "triangle"}
+      ]
+
+      vl = VegaLite.new(title: "Polar Plot", height: 500, width: 500)
+      vl_grid =  VegaLite.Data.polar_grid(vl, [0, 1, 2, 3, 4, 5, 7], angle_marks: [0, 15, 30, 45, 60, 90, 180, 270, 360])
+
+      for {data, mark} <- list, reduce: vl_grid do
+        v ->
+          VegaLite.Data.polar_plot(v, data, mark, r: "r", theta: "theta", color: color_key)
+      end
+  """
+  def polar_plot(vl \\ Vl.new(), data, mark, fields) do
+    vl_polar_config =
+      case vl.spec do
+        %{"_vl_polar_config" => conf} ->
+          Enum.to_list(conf)
+
+        _ ->
+          [radius_marks: nil, angle_offset: 0, direction: :counter_clockwise]
+      end
+
+    data_layer =
+      polar_plot_data_layers(
+        data,
+        mark,
+        fields,
+        vl_polar_config
+      )
+
+    append_layers(vl, [data_layer])
+  end
+
+  defp append_layers(vl, layers) do
+    layers = Enum.map(layers, &(&1 |> Vl.to_spec() |> Map.delete("$schema")))
+
+    update_in(vl.spec, fn spec ->
+      Map.update(spec, "layer", layers, &(&1 ++ layers))
+    end)
+  end
+
+  defp deg_to_rad(angle), do: angle * :math.pi() / 180
+
+  defp polar_angle_layers(opts) do
+    angle_marks_input = opts[:angle_marks]
+    angle_offset = opts[:angle_offset]
+
+    {angle_marks, angle_marks2, angle_offset} =
+      case opts[:direction] do
+        :clockwise ->
```

Comment:
Why have a direction as configuration? Why not allow the angle marks to be in any direction?

Reply:
Direction determines whether the angles will grow clockwise or counterclockwise.

Counterclockwise is the natural mathematical direction.
Clockwise is natural for sailboat polars, for instance (in that case the 0º mark will be at 12 o'clock  instead of the usual 3 o'clock too, which is achieved through :angle_offset).